foundry.md

```diff
@@ -334,18 +329,14 @@ This rule then takes the `uint256` value using `#asWord(#range(LM, ARGSTART +Int
 function chainId(uint256) external;
 ```
 
-`call.chainId` will match when the `chanId` function is called at the [Foundry cheatcode address](https://book.getfoundry.sh/cheatcodes/#cheatcodes-reference).
-This rule then takes the `uint256` value using `#asWord(#range(LM, ARGSTART +Int 4, 32)` and updates the `<chainID>` cell.
+`foundry.call.chainId` will match when the `chainId` cheat code function is called.
+This rule then takes the `uint256` value from the function call data which is represented as `ARGS` and updates the `<chainID>` cell.
 
-```k
-    rule [call.chainId]:
-         <k> CALL _ CHEAT_ADDR _VALUE ARGSTART _ARGWIDTH _RETSTART _RETWIDTH => #finishCheatCode ... </k>
+```{.k .bytes}
+    rule [foundry.call.chainId]:
+         <k> #call_foundry 1078582738 ARGS => . ... </k>
```

Comment:
We can't use `selector("chainId(uint256"))` here?

What if instead we do:

```
rule <k> #call_foundry FID ARGS => . ... </k>
      REST OF RULE
  requires FID ==Int selector("chainId(uint256)")
```

It just seems more readable if we can use the function selector.

But at the very least, we should have a comment saying that this Int is equal to `selector("chainId(uint256)"`

Reply:
I left a comment here regarding the use of the `selector` function https://github.com/runtimeverification/evm-semantics/issues/1405#issuecomment-1260769951 . I'll give your suggestion a try.